### ISSUE TITLE
WFS server filter parameter using <ogc:GmlObjectId> or <ogc:ResourceId> doesn't like feature type name containing dots

### ISSUE THREAD

**Issue by kvlahrogut** [mapserver/mapserver#5741]
System:
- SUSE Linux Enterprise Server 12 Service Pack 4
- x86_64
- Kernel 4.12.14-95.3
- MapServer 7.2.1 (OUTPUT=PNG OUTPUT=JPEG OUTPUT=KML SUPPORTS=PROJ SUPPORTS=AGG SUPPORTS=FREETYPE SUPPORTS=CAIRO SUPPORTS=SVG_SYMBOLS SUPPORTS=RSVG SUPPORTS=ICONV SUPPORTS=WMS_SERVER SUPPORTS=WMS_CLIENT SUPPORTS=WFS_SERVER SUPPORTS=WFS_CLIENT SUPPORTS=WCS_SERVER SUPPORTS=SOS_SERVER SUPPORTS=FASTCGI SUPPORTS=GEOS SUPPORTS=PBF INPUT=JPEG INPUT=POSTGIS INPUT=OGR INPUT=GDAL INPUT=SHAPEFILE)

***

Description:
Thanks to the fix implemented to solve issue #4925, we can have feature type names containing dots and WFS server parameter `featureid`, however, works perfectly well. The GET request …

`https://geo.sv.rostock.de/geodienste/schulen/wfs?service=WFS&version=1.1.0&request=GetFeature&typename=schulen:hro.schulen.grundschulen&featureid=hro.schulen.grundschulen.0ac5533c-cd47-11e4-b1ea-0050569b7e95`

… for example returns exactly the feature we want it to return: the feature with id `0ac5533c-cd47-11e4-b1ea-0050569b7e95` of the feature type `hro.schulen.grundschulen`.

**But:** If we change the link so that it uses WFS server parameter `filter` with `<ogc:GmlObjectId>` instead of `featureid` …

`https://geo.sv.rostock.de/geodienste/schulen/wfs?service=WFS&version=1.1.0&request=GetFeature&typename=schulen:hro.schulen.grundschulen&filter=<ogc:Filter><ogc:GmlObjectId gml:id="hro.schulen.grundschulen.0ac5533c-cd47-11e4-b1ea-0050569b7e95"/></ogc:Filter>`

… MapServer throws an exception:

`FLTPreParseFilterForAlias(): General error message. Feature id hro.schulen.grundschulen.0ac5533c-cd47-11e4-b1ea-0050569b7e95 not consistent with feature type name hro.schulen.grundschulen.`

By the way: exactly the same behaviour with `<ogc:ResourceId>` (FE 2.0 using WFS version 2.0.0).

And since WFS server parameter `filter` with `<ogc:GmlObjectId>` works like a charm, too, if we change feature type name `hro.schulen.grundschulen` to something like, for instance, `hro_schulen_grundschulen`, there seems to be something wrong with MapServer's parsing of `gml:id="…"` if the feature type name is containing dots.

**Comment by kvlahrogut** [comment]
Maybe having a closer look at the function …

https://github.com/mapserver/mapserver/blob/45dda69a25d701891d4ae815d4f2f4fa1baf638d/mapogcfilter.c#L3683

… would lead to a possible solution.

**Comment by ravhed** [comment]
I believe this issue can be closed now since it is merged into branch-7.4. Github says it will "close once commit b8139f2 is merged into the 'branch-7.4' branch'" but seems like it didn't do it for some reason. Can you close it @geographika?

**Comment by geographika** [comment]
@kvlahrogut - if you get a chance to confirm the issue is fixed in your example (using the latest 7.4 release candidate) that would be helpful. Please reopen the issue if it is not.

**Comment by kvlahrogut** [comment]
@ravhed – Thank you very much for the fix! @geographika – I can confirm that feature names with dots are now supported for feature id filtering.
